File: Src/Icm.Core/Compilation/CompileException.vb
Imports System
Imports System.CodeDom.Compiler
Imports System.Runtime.Serialization

Namespace Icm.Compilation

    <Serializable()>
    Public Class CompileException
        Inherits Exception

        Public ReadOnly ErrorList As CompilerErrorCollection

        Public Sub New()
            MyBase.New("Error compiling expression")
            ErrorList = New CompilerErrorCollection
        End Sub

        Public Sub New(ByVal errorList As CompilerErrorCollection, ByVal inner As Exception)
            MyBase.New("Error compiling expression", inner)

            Me.ErrorList = errorList
        End Sub
        Public Sub New(ByVal message As String)
            MyBase.New(message)
            ErrorList = New CompilerErrorCollection
        End Sub
        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)
            ErrorList = New CompilerErrorCollection
        End Sub
        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
            ErrorList = New CompilerErrorCollection
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/Compilation/CompileException.cs
using System;
using System.CodeDom.Compiler;
using System.Runtime.Serialization;

namespace Icm.Compilation
{

	[Serializable]
	public class CompileException : Exception
	{


		public CompilerErrorCollection ErrorList { get; }
		public CompileException() : base("Error compiling expression")
		{
			ErrorList = new CompilerErrorCollection();
		}

		public CompileException(CompilerErrorCollection errorList, Exception inner) : base("Error compiling expression", inner)
		{

			this.ErrorList = errorList;
		}
		public CompileException(string message) : base(message)
		{
			ErrorList = new CompilerErrorCollection();
		}
		public CompileException(string message, Exception innerException) : base(message, innerException)
		{
			ErrorList = new CompilerErrorCollection();
		}
		protected CompileException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
			ErrorList = new CompilerErrorCollection();
		}

	}

}